lua/nvim-treesitter/query.lua

```diff
@@ -120,6 +185,39 @@ function M.iter_prepared_matches(query, qnode, bufnr, start_row, end_row)
           if pred[1] == "set!" and type(pred[2]) == "string" then
             insert_to_path(prepared_match, split(pred[2]), pred[3])
           end
+          if pred[1] == "strip!" and #pred == 3 then
+            local query_name = query.captures[pred[2]]
+            local regex = pred[3]
+
+            local node = utils.get_at_path(prepared_match, query_name..'.node')
+            local ts_utils = require 'nvim-treesitter.ts_utils'
+            local node_lines = ts_utils.get_node_text(node, bufnr)
+            local start_line, start_col, end_line, end_col = node:range()
+
+            local strip_line, strip_col = strip_beginning(node_lines, regex)
+            start_line = start_line + strip_line
+            if strip_line == 0 then
+              start_col = start_col + strip_col
+            else
+              start_col =strip_col
+            end
+            local strip_line, strip_col = strip_end(node_lines, regex)
+            end_line = end_line - strip_line
+            if strip_line == 0 then
+              end_col = end_col - strip_col
+            else
+              end_col = strip_col
+            end
```

Comment:
That name redefinition makes things less clear, I think that luacheck will be triggered by this

Reply:
Actually not, this is the way luacheck likes. For the redefinition of the local as @kyazdani42 does it all over the repo I had to disable the warning.

Strip transforms the range so I found it natural to modify the range variables. But I can rename it to processed_end_line.

Reply:
Ok, that's fine for me